model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -2198,7 +2169,33 @@ async def execute(self, user: User, request: ModelDownloadRequest) -> ModelDownl
         return ModelDownloadResponse(urls=urls)
 
 
-def infer_hardware_from_model_name(model_name: str) -> CreateDockerImageBatchJobResourceRequests:
+def _fill_hardware_info(request: CreateLLMModelEndpointV1Request):
+    if (
+        request.gpus is None
+        or request.gpu_type is None
+        or request.cpus is None
+        or request.memory is None
+        or request.storage is None
+    ):
+        if not (
+            request.gpus is None
+            and request.gpu_type is None
+            and request.cpus is None
+            and request.memory is None
+            and request.storage is None
+        ):
+            raise ObjectHasInvalidValueException(
+                "All hardware spec fields (gpus, gpu_type, cpus, memory, storage) must be provided if any hardware spec field is provided."
```

Comment:
I'm not sure if this sentence aligns with the logic?

```python
if not (
            request.gpus is None
            and request.gpu_type is None
            and request.cpus is None
            and request.memory is None
            and request.storage is None
        ):
```
seems to align with `All hardware spec fields [...] must be provided [...]` but

```python
 if (
        request.gpus is None
        or request.gpu_type is None
        or request.cpus is None
        or request.memory is None
        or request.storage is None
    ):
```
Seems to say "if any of these fields is missing", not "if any of these fields is provided".

Reply:
yeah good catch